Bourreau/spec/factories/bourreau_factories.rb

```diff
@@ -0,0 +1,270 @@
+  factory :data_provider do
+    sequence(:name) { |n| "dataprovider_#{n}" }
+    read_only       true
+    type "FlatDirLocalDataProvider"
+    association     :user, factory: :normal_user
+    association     :group
+  end
+
+  factory :ssh_data_provider, parent: :data_provider, class: SshDataProvider do
+    sequence(:name) { |n| "ssh_dataprovider_#{n}" }
+  end
+
+  factory :vault_smart_data_provider, parent: :data_provider, class: VaultSmartDataProvider do
+    sequence(:name) { |n| "vault_smart_dataprovider_#{n}" }
+  end
+
+  factory :vault_ssh_data_provider, parent: :data_provider, class: VaultSshDataProvider do
+    sequence(:name) { |n| "vault_ssh_dataprovider_#{n}" }
+  end
+
+  factory :vault_local_data_provider, parent: :data_provider, class: VaultLocalDataProvider do
+    sequence(:name) { |n| "vault_local_dataprovider_#{n}" }
+  end
+
+  factory :flat_dir_local_data_provider, parent: :data_provider, class: FlatDirLocalDataProvider do
+    sequence(:name) { |n| "local_dataprovider_#{n}" }
+  end
+
+  factory :en_cbrain_smart_data_provider, parent: :data_provider, class: EnCbrainSmartDataProvider do
+    sequence(:name) { |n| "en_cb_smart_dataprovider_#{n}" }
+  end
+
+  factory :en_cbrain_ssh_data_provider, parent: :data_provider, class: EnCbrainSshDataProvider do
+    sequence(:name) { |n| "en_cb_ssh_dataprovider_#{n}" }
+  end
+
+  factory :en_cbrain_local_data_provider, parent: :data_provider, class: EnCbrainLocalDataProvider do
+    sequence(:name) { |n| "en_cb_local_dataprovider_#{n}" }
+  end
+
+  factory :incoming_vault_ssh_data_provider, parent: :data_provider, class: IncomingVaultSshDataProvider do
+    sequence(:name) { |n| "in_vault_ssh_dataprovider_#{n}" }
+  end
+
+
+
+  #################
+  # Userfile      #
+  #################
+
+  factory :userfile do
+    sequence(:name) { |n| "file_#{n}" }
+    association     :user, factory: :normal_user
+    association     :group
+    association     :data_provider
+  end
+
+  Userfile.nil? # force pre-load of all constants under Userfile
+  factory :text_file, parent: :userfile, class: TextFile do
+    sequence(:name) { |n| "text_file_#{n}" }
+  end
+
+  factory :single_file, parent: :userfile, class: SingleFile do
+    sequence(:name) { |n| "single_file_#{n}" }
+  end
+
+  factory :file_collection, parent: :userfile, class: FileCollection do
+    sequence(:name) { |n| "file_collection_#{n}" }
+  end
+
+
+
+  ###################
+  # Remote Resource #
+  ###################
+
+  factory :remote_resource do
+    sequence(:name)    { |n| "rr_#{n}" }
+    online             true
+    dp_ignore_patterns ["x", "y", "z"]
+    association        :user, factory: :normal_user
+    association        :group
+  end
+
+  factory :brain_portal, parent: :remote_resource, class: BrainPortal do
+    sequence(:name) { |n| "bp_#{n}"}
+  end
+
+  factory :bourreau, parent: :remote_resource, class: Bourreau do
+    sequence(:name) { |n| "bourreau_#{n}"}
+    cms_class       "ScirSge"
+  end
+
+
```

Comment:
This file is a copy of the portal_factories without the CustomFilter block. If I well understand you only did that because Custom filters are not defined on the Bourreau side. 

I personally will create a symlink `bourreau_factories.rb` that point to `portal_factories` and add a `begin .. rescue` block where needed. Like that 

```ruby 
  begin
    factory :userfile_custom_filter, parent: :custom_filter, class: UserfileCustomFilter do
      sequence(:name) { |n| "ucf_#{n}"}
    end
  rescue
    puts "On the Bourreau side of test, we do not need to create Userfile custom_filter"
  end

  begin
    factory :task_custom_filter, parent: :custom_filter, class: TaskCustomFilter do
      sequence(:name) { |n| "tcf_#{n}"}
    end
  rescue
    puts "On the Bourreau side of test, we do not need to create Task custom_filter"
  end
```